### PR TITLE
TST: make get_upcast_box more flexible

### PR DESCRIPTION
**Pull request by jbrockmendel**
Will make it easier to parametrize these tests

### CODE REVIEW

**Comment by jreback** [comment]
ok can we disable the sdist 3.10 test :->

**Comment by jbrockmendel** [comment]
> ok can we disable the sdist 3.10 test :->

sure but separate PR

File: pandas/_testing/__init__.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def box_expected(expected, box_cls, transpose=True):
         expected = DatetimeArray(expected)
     elif box_cls is TimedeltaArray:
         expected = TimedeltaArray(expected)
-    elif box_cls is np.ndarray:
+    elif box_cls is np.ndarray or box_cls is np.array:
```
Comment (jreback) [inline pandas/_testing/__init__.py:262]: umm this is very strange

Comment (jbrockmendel) [reply pandas/_testing/__init__.py:262]: we're a bit inconsistent about when we use the class vs the constructor; np.ndarray(foo) doesn't work like np.array(foo)

Comment (jreback) [reply pandas/_testing/__init__.py:262]: ok sure, but should ideally fix this at some point.

**Pull request merged**